apps/root/src/services/aggregatorService.ts

```diff
@@ -91,28 +91,45 @@ export default class AggregatorService {
     return this.safeService.submitMultipleTxs([approveTx, route.tx as TransactionRequest]);
   }
 
-  async getSwapOptions(
-    from: Token,
-    to: Token,
-    sellAmount?: bigint,
-    buyAmount?: bigint,
-    sorting?: SwapSortOptions,
-    transferTo?: string | null,
-    slippage?: number,
-    gasSpeed?: GasKeys,
-    takerAddress?: Address,
-    chainId?: number,
-    disabledDexes?: string[],
+  async getSwapOptions({
+    from,
+    to,
+    sellAmount,
+    buyAmount,
+    sorting,
+    transferTo,
+    slippage,
+    gasSpeed,
+    takerAddress,
+    chainId,
+    disabledDexes,
     usePermit2 = false,
-    sourceTimeout = TimeoutKey.patient
-  ) {
-    const currentNetwork = await this.providerService.getNetwork(takerAddress);
+    sourceTimeout = TimeoutKey.patient,
+  }: {
+    from: Token;
+    to: Token;
+    chainId: number;
+    sellAmount?: bigint;
+    buyAmount?: bigint;
+    sorting?: SwapSortOptions;
+    transferTo?: string | null;
+    slippage?: number;
+    gasSpeed?: GasKeys;
+    takerAddress?: Address;
+    disabledDexes?: string[];
+    usePermit2?: boolean;
+    sourceTimeout?: TimeoutKey;
+  }) {
     const balance = await this.walletService.getBalance({ account: takerAddress, token: from });
 
-    const isOnNetwork = !chainId || currentNetwork.chainId === chainId;
+    let isOnNetwork = false;
+    try {
+      const currentNetwork = await this.providerService.getNetwork(takerAddress);
+      isOnNetwork = !chainId || currentNetwork.chainId === chainId;
```

Comment:
It is worse for readability, but if `!chainId`, then you don't need to ask the provider service for the network, and you don't need to wait for it
```suggestion
      isOnNetwork = !chainId || (await this.providerService.getNetwork(takerAddress)).chainId === chainId;
```

Reply:
Agree

Reply:
Like it!